### PR TITLE
mockgcp: support for projects

### PR DESCRIPTION
**Pull request by justinsb**
Add support for projects.  Some complexity because the version of projects terraform uses doesn't have a published proto.  Also implement a bit of the billing because projects try to set billing.

### CODE REVIEW

**Comment by justinsb** [comment]
Likely blocked on #911 (includes it as the first 2 commits!)

**Comment by justinsb** [comment]
No longer blocked on #911, as that was merged (thanks @gemmahou )

Marking as no longer WIP

File: pkg/test/gcp/gcp.go [hunk 1]
```diff
@@ -52,15 +52,17 @@ func (v *EnvVar) Set(s string) {
 }
 
 var (
+	TestFolderID                        = EnvVar{Key: "TEST_FOLDER_ID"}
+	TestFolder2ID                       = EnvVar{Key: "TEST_FOLDER_2_ID"}
 	TestOrgID                           = EnvVar{Key: "TEST_ORG_ID"}
 	TestBillingAccountID                = EnvVar{Key: "TEST_BILLING_ACCOUNT_ID"}
 	IAMIntegrationTestsOrganizationID   = EnvVar{Key: "IAM_INTEGRATION_TESTS_ORGANIZATION_ID"}
 	IAMIntegrationTestsBillingAccountID = EnvVar{Key: "IAM_INTEGRATION_TESTS_BILLING_ACCOUNT_ID"}
+	TestAttachedClusterName             = EnvVar{Key: "TEST_ATTACHED_CLUSTER_NAME"}
```
Comment (gemmahou) [inline pkg/test/gcp/gcp.go:61]: I think I forgot to delete this variable. This is no longer used anywhere.

File: tests/e2e/unified_test.go [hunk 2]
```diff
@@ -41,31 +40,50 @@ func TestAllInSeries(t *testing.T) {
...
-		}
 		// Some fixed-value fake org-ids for testing.
 		// We used fixed values so that the output is predictable (for golden testing)
+		testgcp.TestFolderID.Set("123451001")
+		testgcp.TestFolder2ID.Set("123451002")
 		testgcp.TestOrgID.Set("123450001")
 		testgcp.TestBillingAccountID.Set("123456-777777-000001")
 		testgcp.IAMIntegrationTestsOrganizationID.Set("123450002")
 		testgcp.IAMIntegrationTestsBillingAccountID.Set("123456-777777-000002")
+		testgcp.TestAttachedClusterName.Set("xks-cluster")
```
Comment (gemmahou) [inline tests/e2e/unified_test.go:61]: We can remove this too.

File: mockgcp/mockbilling/names.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+		name := &projectName{
+			Project: project,
+		}
+
+		return name, nil
+	} else {
+		return nil, status.Errorf(codes.InvalidArgument, "name %q is not valid", name)
+	}
+}
```
Comment (gemmahou) [inline mockgcp/mockbilling/names.go:53]: Maybe it's better to move above functions into mockresourcemanager/names.go?

File: pkg/test/gcp/gcp.go (hunk 1 above)
Comment (justinsb) [reply pkg/test/gcp/gcp.go:61]: Thanks - removed!

File: tests/e2e/unified_test.go (hunk 2 above)
Comment (justinsb) [reply tests/e2e/unified_test.go:61]: Done!

File: pkg/test/gcp/gcp.go (hunk 1 above)
Comment (justinsb) [reply pkg/test/gcp/gcp.go:61]: Actually, looks like we still use ATTACHED_CLUSTER_NAME in the samples (which we also run / can run in the unified tests)

File: mockgcp/mockbilling/names.go (hunk 3 above)
Comment (justinsb) [reply mockgcp/mockbilling/names.go]: Good call - done!

File: pkg/test/gcp/gcp.go (hunk 1 above)
Comment (gemmahou) [reply pkg/test/gcp/gcp.go:61]: Samples test for this resource is disabled: https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/config/tests/samples/create/samples_test.go#L69-L73.

**Review by gemmahou — APPROVED** [review]

**Pull request merged**